src/main/java/seedu/meal360/Meal360.java

```diff
@@ -0,0 +1,297 @@
+                    break;
+                }
+            } catch (IllegalArgumentException | InvalidNegativeValueException | InvalidRecipeNameException |
+                     ArrayIndexOutOfBoundsException e) {
+                ui.printMessage(e.getMessage());
+            }
+            ui.printSeparator();
+        } else if (command[0].equals("weeklydone")) {
+            ui.printSeparator();
+            try {
+                parser.parseMarkDone(command, userIngredients, weeklyPlan, recipeList);
+                ui.printMessage("I've recorded that you've done this recipe!");
+                ui.printMessage("Ingredients list has been updated accordingly!");
+            } catch (IllegalArgumentException | ArrayIndexOutOfBoundsException |
+                     IngredientNotFoundException e) {
+                ui.printMessage(e.getMessage());
+            }
+        } else if (command[0].equals("available")) {
+            // list recipes with ingredients all in ingredient list
+            ui.printSeparator();
+            RecipeList availableRecipes = recipeList.availableRecipes();
+            ui.listAvailableRecipes(availableRecipes);
+            ui.printSeparator();
+        } else if (command[0].equals("weeklyingredients")) {
+            ui.printSeparator();
+            ui.printWeeklyIngredients(weeklyPlan, recipeList);
+            ui.printSeparator();
+        } else if (command[0].equals("weeklyplan")) {
+            ui.printSeparator();
+            ui.printWeeklyPlan(weeklyPlan);
+            ui.printSeparator();
+        } else if (command[0].equals("add_i")) {
+            ui.printSeparator();
+            try {
+                parser.parseAddUserIngredients(command, userIngredients);
+                ui.printMessage("Ingredient successfully added!");
+            } catch (IllegalArgumentException | IndexOutOfBoundsException e) {
+                ui.printMessage(e.getMessage());
+            }
+            ui.printSeparator();
+        } else if (command[0].equals("del_i")) {
+            ui.printSeparator();
+            try {
+                parser.parseDeleteUserIngredients(command, userIngredients);
+                ui.printMessage("Ingredients successfully deleted!");
+            } catch (IllegalArgumentException | IndexOutOfBoundsException | IngredientNotFoundException e) {
+                ui.printMessage(e.getMessage());
+            }
+            ui.printSeparator();
+        } else if (command[0].equals("view_ingredients")) {
+            ui.printSeparator();
+            ui.printUserIngredients(userIngredients);
+            ui.printSeparator();
+        } else if (command[0].equals("help")) {
+            ui.printSeparator();
+            ui.printHelp();
+            ui.printSeparator();
+        } else {
+            ui.printSeparator();
+            ui.printMessage("I'm sorry, but I don't know what that means :-(");
+            ui.printSeparator();
+        }
+    }
```

Comment:
method seems to be very long. Do consider grouping related behaviour and refactoring it into smaller methods to reduce its LOC and to make it more concise